src/Accounts/Authentication/Factories/AuthenticationFactory.cs

```diff
@@ -462,14 +467,18 @@ private string GetFunctionsResourceId(string resourceIdOrEndpointName, IAzureEnv
         private string GetEndpointToken(IAzureAccount account, string targetEndpoint)
         {
             string tokenKey = AzureAccount.Property.AccessToken;
-            if (string.Equals(targetEndpoint, AzureEnvironment.Endpoint.Graph, StringComparison.OrdinalIgnoreCase))
-            {
-                tokenKey = AzureAccount.Property.GraphAccessToken;
-            }
             if (string.Equals(targetEndpoint, AzureEnvironment.Endpoint.AzureKeyVaultServiceEndpointResourceId, StringComparison.OrdinalIgnoreCase))
             {
                 tokenKey = AzureAccount.Property.KeyVaultAccessToken;
             }
+            if (string.Equals(targetEndpoint, AzureEnvironment.ExtendedEndpoint.MicrosoftGraphEndpointResourceId, StringComparison.OrdinalIgnoreCase))
+            {
+                tokenKey = account.GetProperty(Constants.MicrosoftGraphAccessToken);
```

Comment:
```suggestion
                tokenKey = Constants.MicrosoftGraphAccessToken;
```
`Constants.MicrosoftGraphAccessToken` is the key. You get the value in line 482. Isn't it?

Reply:
Good catch.